### PR TITLE
Issue 3702: Prevent suppression of stack traces in tests

### LINKED ISSUE #3702: Prevent suppression of stack traces in tests
**Problem description**
Sometimes tests have the following pattern:
```
try {
 // some operation
} catch (SomeException e) {
    Assert.fail("should not get an exception");
}
```
This is unnecessarily verbose, but also prevents the stack trace or message of the exception from being logged. If the exception is unexpected this is critical debugging information that we could miss. 

**Problem location**
Numerous files

**Suggestions for an improvement**
Remove the try catch and just throw the exception.

### PR DESCRIPTION
**Pull request by tkaitchuck**
**Change log description**  
* Prevent tests from suppressing exception stack traces 

**Purpose of the change**  
Fixes #3702

**What the code does**  
Removes unneeded try catch code from tests. No functional changes.

### CODE REVIEW

**Comment by fpj** [comment]
I have posted a comment in the issue, to see if we can agree on the guidelines for handling exceptions in test code.

File: controller/src/test/java/io/pravega/controller/server/PravegaTablesControllerServiceMainTest.java [hunk 1]
```diff
@@ -104,46 +92,22 @@ public void testZKSessionExpiry() throws Exception {
...
-            Assert.fail("Failed waiting for controllerServiceStarter to get ready");
-            return;
-        }
+        controllerServiceStarter.awaitRunning();
 
         assertEquals(1, clientQueue.size());
         CuratorFramework curatorClient = (CuratorFramework) clientQueue.poll().getClient();
         // Simulate ZK session timeout
         // we will submit zk session expiration and 
-        CompletableFuture.runAsync(() -> {
```
Comment (shiveshr) [inline controller/src/test/java/io/pravega/controller/server/PravegaTablesControllerServiceMainTest.java]: this was originally injecting the session expiration and making the zk client call asynchronously. 

although looking at the code, it doesnt make difference.

File: controller/src/test/java/io/pravega/controller/server/ControllerServiceStarterTest.java [hunk 2]
```diff
@@ -53,34 +53,21 @@
...
         // Now, that starter has started, perform some rpc operations.
-        URI uri;
-        try {
-            uri = new URI( (enableAuth ? "tls" : "tcp") + "://localhost:" + grpcPort);
-        } catch (URISyntaxException e) {
-            log.error("Error creating controller URI", e);
-            Assert.fail("Error creating controller URI");
-            return;
-        }
+        URI uri = new URI( (enableAuth ? "tls" : "tcp") + "://localhost:" + grpcPort);
```
Comment (RaulGracia) [inline controller/src/test/java/io/pravega/controller/server/ControllerServiceStarterTest.java]: Extra space between parentheses.

Comment (fpj) [reply controller/src/test/java/io/pravega/controller/server/ControllerServiceStarterTest.java]: This is not adding the space, it is from the previous code, no reformatting is required here.

**Pull request merged**